src/sys/system.h

```diff
@@ -152,25 +162,29 @@ class System
      */
     const Config& GetConfig() const { return cfg_; }
 
-    /** Returns an enum representing the current (primary) memory space used
+    /** Returns an enum representing the current (primary) memory space used 
      *  for executing the program.
+     *  \param program_start Optionally indicate a specific address to be checked. 
+     *  Checks the VTOR by default, which will generally indicate the memory
+     *  from which the program is executed.
      */
-    static ProgramMemory GetProgramMemory();
+    static ProgramMemory GetProgramMemory(uint32_t program_start = SCB->VTOR);
+
+    static constexpr uint32_t kSramStart = 0x24000000U;
+    static constexpr uint32_t kSramEnd   = kSramStart + 0x80000U;
+    static constexpr uint32_t kQspiStart = 0x90040000U;
+
+    static constexpr uint32_t kQspiEnd       = kQspiStart + 0x7c0000;
+    static constexpr uint32_t kInternalStart = 0x08000000U;
+    static constexpr uint32_t kInternalEnd   = kInternalStart + 0x20000U;
+
+    static constexpr uint32_t kExpectedStack = 0x20020000U;
```

Comment:
Is this const used for anything? 

idk if we should expect the stack to be in one place or the other because A: we just made the default linker conditional. So users can move it elsewhere. and B: if we wanted access to this we could probably provide/wrap access around a linker symbol instead.

Reply:
Maybe I'm just bad, but I didn't find the linker variable approach very straightforward. Even after declaring the correct variables as `extern` within libdaisy, it seemed like the final compilation failed to actually link them correctly.

In any case, that approach wouldn't really work here because this is used by the bootloader to check _other_ binaries. Assuming the stack is located in the same place, checking the first four bytes is a convenient and reliable way to determine if a binary is likely correctly formatted and executable. I suppose that checking could be skipped, though. As long as the stack and entry point are within valid memory, I suppose it's fine.

Reply:
you're not bad, haha. There's probably something weird about them that makes it trickier to deal with.  I think I had something reading one of them at some point a long time ago when I was doing some QSPI prior to the daisy release, but I can't remember if/what I had to do to actually read them.